src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/stdx/string_view.hpp

```diff
@@ -14,92 +14,489 @@
+        return size() == 0;
+    }
+    /// Return the maximum value that could be returned by size()
+    constexpr size_type max_size() const noexcept {
+        return static_cast<size_type>(std::numeric_limits<difference_type>::max());
+    }
+
+    /**
+     * @brief In-place modify the string_view to view N fewer characters from the beginning
+     *
+     * @param n The number of characters to remove from the beginning. Must be less than size()
+     */
+    bsoncxx_cxx14_constexpr void remove_prefix(size_type n) noexcept {
+        _begin += n;
+        _size -= n;
+    }
+
+    /**
+     * @brief In-place modify the string_view to view N fewer characters from the end
+     *
+     * @param n The number of characters to remove from the end. Must be less than size()
+     */
+    bsoncxx_cxx14_constexpr void remove_suffix(size_type n) noexcept {
+        _size -= n;
+    }
+
+    /**
+     * @brief Swap the reference with another string_view
+     */
+    bsoncxx_cxx14_constexpr void swap(basic_string_view& other) noexcept {
+        std::swap(_begin, other._begin);
+        std::swap(_size, other._size);
+    }
+
+    /**
+     * @brief Copy the contents of the viewed string into the given output destination.
+     *
+     * @param dest The destination at which to write characters
+     * @param count The maximum number of characters to copy.
+     * @param pos The offset within the viewed string to begin copying from.
+     * @returns The number of characters that were copied to `dest`. The number
+     * of copied characters is always the lesser of `size()-pos` and `count`
+     *
+     * @throws std::out_of_range if pos > size()
+     */
+    bsoncxx_cxx14_constexpr size_type copy(pointer dest, size_type count, size_type pos = 0) const {
+        if (pos > size()) {
+            throw std::out_of_range{"bsoncxx::stdx::basic_string_view::substr()"};
+        }
+        count = (std::min)(count, size() - pos);
+        Traits::copy(dest, data() + pos, count);
+        return count;
+    }
+
+    /**
+     * @brief Obtain a substring of this string
+     *
+     * @param pos The zero-based index at which to start the new string.
+     * @param count The number of characters to include following `pos` in the new string.
+     * Automatically clamped to the available size
+     *
+     * @throws std::out_of_range if `pos` is greater than this->size()
+     */
+    bsoncxx_cxx14_constexpr self_type substr(size_type pos, size_type count = npos) const {
+        if (pos > size()) {
+            throw std::out_of_range{"bsoncxx::stdx::basic_string_view::substr()"};
+        }
+        return self_type(_begin + pos, (std::min)(count, size() - pos));
+    }
+
+    /**
+     * @brief Compare two strings lexicographically
+     *
+     * @param other The "right hand" operand of the comparison
+     * @retval 0 If *this == other
+     * @retval n : n < 0 if *this is "less than" other.
+     * @retval n : n > 0 if *this is "greater than" other.
```

Comment:
Doxygen doesn't seem to like repeated use of the `@retval` command with the same return value:

```
warning: return value 'n' of bsoncxx::v_noabi::stdx::basic_string_view::compare has multiple documentation sections
warning: return value 'n' of bsoncxx::v_noabi::stdx::basic_string_view::compare has multiple documentation sections
```

Suggest either:

```
@retval negative-value if `*this` is "less than" other.
@retval positive-value if `*this` is "less than" other.
```
Or using `@return` instead (no return value argument that demands uniqueness):
```
* @return `0` if `*this == other`.
* @return `n : n < 0` if `*this` is "less than" other.
* @return `n : n > 0` if `*this` is "greater than" other.
```